Objects/unicodeobject.c

```diff
@@ -8594,7 +8606,7 @@ charmapencode_output(Py_UCS4 c, PyObject *mapping,
         return enc_SUCCESS;
     }
 
-    rep = charmapencode_lookup(c, mapping);
+    rep = charmapencode_lookup(c, mapping, &replace);
```

Comment:
You can move `unsigned char replace;` declaration here, to reduce the scope of the variable.

Reply:
I added it next to the declaration of `rep`, as in all other cases.